gensim/models/doc2vec.py

```diff
@@ -636,6 +636,12 @@ def scan_vocab(self, documents, progress_per=10000, trim_rule=None):
         interval_start = default_timer() - 0.00001  # guard against next sample being identical
         interval_count = 0
         vocab = defaultdict(int)
+        try:
+            if isinstance(list(documents)[0].words, string_types):
```

Comment:
On the existing iteration, check the current sentence. *But*, once the first is checked, set a flag to prevent further checks – based on our assumption that the user either gets this "all right" or "all wrong" and that (when wrong) they don't want redundant warnings for every sentence.